sdk/src/Core/Amazon.Runtime/Pipeline/HttpHandler/_netstandard/HttpRequestMessageFactory.cs

```diff
@@ -625,13 +625,10 @@ public void WriteToRequestBody(HttpContent requestContent, Stream contentStream,
             if ((isChunkedUploadWrapperStreamWithLength || isTrailingHeadersWrapperStreamWithLength || isCompressionWrapperStreamWithLength)
                 || (chunkedUploadWrapperStream == null && trailingHeadersWrapperStream == null && compressionWrapperStream == null))
             {
-                long position = 0;
+                long position;
                 try
                 {
-                    if (contentStream.CanSeek)
-                    {
-                        position = contentStream.Position;
-                    }
+                    position = contentStream.Position;
```

Comment:
nit: Not relying on the `CanSeek` is due to special circumstances of our `ChunkedUploadWrapperStream`. You might want to add a comment here so somebody in the future, like myself, don't suggest adding back the CanSeek check to avoid the control flow by exception.